### PR TITLE
fix(GlobalHeader): Accessibility & Tests Fix

### PR DESCRIPTION
**Pull request by thedoomshine**
### PR Checklist

- [ ] Related to designs:
- [ ] Related to JIRA ticket: ABC-123
- [ ] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@6.1.0-alpha.8aac38.0<br/>@codecademy/gamut-kit@0.6.38-alpha.8aac38.0<br/>@codecademy/gamut-labs@28.6.1-alpha.8aac38.0<br/>@codecademy/gamut@42.3.0-alpha.8aac38.0<br/>@codecademy/monaco@0.5.28-alpha.8aac38.0<br/>@codecademy/styleguide@52.5.0-alpha.8aac38.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61d5ef8795fdea9fe8711cb2--gamut-preview.netlify.app>https://61d5ef8795fdea9fe8711cb2--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61d5ef8795fdea9fe8711cb2>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@6.1.0-alpha.d6f467.0<br/>@codecademy/gamut-kit@0.6.38-alpha.d6f467.0<br/>@codecademy/gamut-labs@28.6.1-alpha.d6f467.0<br/>@codecademy/gamut@42.3.0-alpha.d6f467.0<br/>@codecademy/monaco@0.5.28-alpha.d6f467.0<br/>@codecademy/styleguide@52.5.0-alpha.d6f467.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61d5f28b141e31c05d3a1b00--gamut-preview.netlify.app>https://61d5f28b141e31c05d3a1b00--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61d5f28b141e31c05d3a1b00>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@6.1.0-alpha.7663f7.0<br/>@codecademy/gamut-kit@0.6.38-alpha.7663f7.0<br/>@codecademy/gamut-labs@28.6.1-alpha.7663f7.0<br/>@codecademy/gamut@42.3.0-alpha.7663f7.0<br/>@codecademy/monaco@0.5.28-alpha.7663f7.0<br/>@codecademy/styleguide@52.5.0-alpha.7663f7.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61d7111e56a6f948ca1bce14--gamut-preview.netlify.app>https://61d7111e56a6f948ca1bce14--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61d7111e56a6f948ca1bce14>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@6.1.0-alpha.a8bcdb.0<br/>@codecademy/gamut-kit@0.6.38-alpha.a8bcdb.0<br/>@codecademy/gamut-labs@28.6.1-alpha.a8bcdb.0<br/>@codecademy/gamut@42.3.0-alpha.a8bcdb.0<br/>@codecademy/monaco@0.5.28-alpha.a8bcdb.0<br/>@codecademy/styleguide@52.5.0-alpha.a8bcdb.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61d7167b5b1ce94585ed0674--gamut-preview.netlify.app>https://61d7167b5b1ce94585ed0674--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61d7167b5b1ce94585ed0674>Deploy Logs</a></p>

File: packages/gamut/src/ButtonBase/ButtonBase.tsx [hunk 1]
```diff
@@ -59,35 +59,32 @@ const ResetElement = styled('button', styledOptions<'button'>())(resetStyles);
...
+  if (href == null || disabled) {
     return (
       <ResetElement
         {...rest}
-        ref={ref as MutableRefObject<HTMLAnchorElement>}
-        as="a"
-        href={href}
+        ref={ref as MutableRefObject<HTMLButtonElement>}
+        as="button"
+        type={type ?? 'button'}
```
Comment (JoshuaKGoldberg) [inline packages/gamut/src/ButtonBase/ButtonBase.tsx:69]: Nit: covered in props?

```suggestion
        type={type}
```

File: packages/gamut/src/ButtonBase/ButtonBase.tsx [hunk 2]
```diff
@@ -59,35 +59,32 @@ const ResetElement = styled('button', styledOptions<'button'>())(resetStyles);
...
+>(({ href, disabled, children, role, type = 'button', ...rest }, ref) => {
+  if (href == null || disabled) {
     return (
       <ResetElement
         {...rest}
-        ref={ref as MutableRefObject<HTMLAnchorElement>}
-        as="a"
-        href={href}
+        ref={ref as MutableRefObject<HTMLButtonElement>}
+        as="button"
```
Comment (JoshuaKGoldberg) [inline packages/gamut/src/ButtonBase/ButtonBase.tsx:68]: This scares me. `ButtonBase` is used by a lot of places, including the shared `createButtonComponent`. In theory I agree with the change _(`a` without `href`? blurgh!)_ but we'd have to do a lot of QA to make sure random weird things in the monolith don't break with it. 

Maybe a followup ticket so the hard QA doesn't block this PR? And whatever needs `ButtonBase` in this PR should provide the `as` explicitly?

File: packages/gamut-labs/src/HeaderHeightArea/index.tsx [hunk 3]
```diff
@@ -31,12 +31,14 @@ const HeaderHeightAreaBase = styled(Box)(
 export type HeaderHeightAreaProps = {
   as?: React.ElementType<any>;
   display: ResponsiveProp<'none' | 'block'>;
+  title?: string;
```
Comment (JoshuaKGoldberg) [inline packages/gamut-labs/src/HeaderHeightArea/index.tsx:34]: Question: this is provided in both places. Should it be required?

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
Tested on https://github.com/Codecademy/gamut/pull/2221 and it looks great, thanks!

Just requesting changes on the `ButtonBase` thing as I am scared

File: packages/gamut/src/ButtonBase/ButtonBase.tsx (hunk 1 above)
Comment (thedoomshine) [reply packages/gamut/src/ButtonBase/ButtonBase.tsx:69]: we want it to default to `role="button"` only if it doesnt have an `href`, if its a link then role isnt necessary and this lets it be set to `undefined`

File: packages/gamut-labs/src/HeaderHeightArea/index.tsx (hunk 3 above)
Comment (thedoomshine) [reply packages/gamut-labs/src/HeaderHeightArea/index.tsx:34]: i have no idea tbh, i think that was a @dreamwasp addition

File: packages/gamut/src/ButtonBase/ButtonBase.tsx (hunk 1 above)
Comment (JoshuaKGoldberg) [reply packages/gamut/src/ButtonBase/ButtonBase.tsx:69]: Agreed - and `type = 'button'` above, so `type` is always going to be `string` on line 69.

File: packages/gamut/src/ButtonBase/ButtonBase.tsx (hunk 2 above)
Comment (JoshuaKGoldberg) [reply packages/gamut/src/ButtonBase/ButtonBase.tsx:68]: Talked on Slack, not a logic change! Yay!

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@6.1.0-alpha.913aa3.0<br/>@codecademy/gamut-kit@0.6.38-alpha.913aa3.0<br/>@codecademy/gamut-labs@28.6.1-alpha.913aa3.0<br/>@codecademy/gamut@42.3.0-alpha.913aa3.0<br/>@codecademy/monaco@0.5.28-alpha.913aa3.0<br/>@codecademy/styleguide@52.5.0-alpha.913aa3.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61d72dea37b10a5f16ad3943--gamut-preview.netlify.app>https://61d72dea37b10a5f16ad3943--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61d72dea37b10a5f16ad3943>Deploy Logs</a></p>

File: packages/gamut-labs/src/HeaderHeightArea/index.tsx (hunk 3 above)
Comment (dreamwasp) [reply packages/gamut-labs/src/HeaderHeightArea/index.tsx:34]: we can nix this, i think i just seeing what stuck when i added this 😅

**Review by JoshuaKGoldberg — APPROVED** [review]
✨

**Review by dreamwasp — APPROVED** [review]
🙏

**Pull request merged**